Pod/Classes/ResultType.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  ResultType.swift
+//  Pods
+//
+//  Created by Bueno McCartney on 12/7/16.
+//
+//
+
+import Foundation
+
+public protocol ResultType {
+    associatedtype Value
+    
+    init(_ value: Value?)
+    init(_ error: Error)
```

Comment:
@codecaffeine @jgrandelli (or anyone) What are the feelings on this being
```
    init(_ value: Value?)
    init(_ error: Error)
```
vs.
```
    init(value: Value?)
    init(error: Error)
```

In the current version we are working with you need to call out "error" or "value" when you are initing the Result which I think in some cases reads a little better.